connectors/http/src/main/java/quarks/connectors/http/HttpStreams.java

```diff
@@ -38,12 +40,88 @@ Licensed to the Apache Software Foundation (ASF) under one
+     *     request1.addProperty("a", "abc");
+     *     request1.addProperty("b", "42");
+     *     TStream<JsonObject> rc = HttpStreams.getJson(
+     *             topology.collection(Arrays.asList(request1)),
+     *             HttpClients::noAuthentication,
+     *             t -> url + "a=" + t.get("a").getAsString() + "&b="
+     *                     + t.get("b").getAsString());
+     * }
+     * </pre>
+     * <br>
+     * See <i>HttpTest</i> for example.
+     * <br>
+     * @param stream - JsonObject TStream.
+     * @param clientCreator - CloseableHttpClient supplier preferably created using {@link HttpClients}
+     * @param uri - URI function which returns URI string
+     * @return TStream of JsonObject which contains responses of GET requests
+     * 
+     * @see HttpStreams#requests(TStream, Supplier, Function, Function, BiFunction)
+     */
     public static TStream<JsonObject> getJson(TStream<JsonObject> stream,
             Supplier<CloseableHttpClient> clientCreator,
             Function<JsonObject,String> uri) {
         
         return HttpStreams.<JsonObject,JsonObject>requests(stream, clientCreator,
-                t -> HttpGet.METHOD_NAME, uri, HttpResponders.json());
+            t -> HttpGet.METHOD_NAME, uri, HttpResponders.json());
+    }
+    
+    /**
+     * Make an HTTP POST request with JsonObject. <br>
+     * 
+     * Method specifically works with JsonObjects. For each JsonObject in the stream, 
+     * HTTP POST request is executed on provided uri. Request body is filled using
+     * HttpEntity provided by body function. As a result, Response is added to
+     * the response TStream.<br>
+     * 
+     * Sample usage:<br>
+     * 
+     * <pre>
+     * {@code
+     *     JsonObject request = new JsonObject();
+     *     request.addProperty("a", "abc");
+     *     request.addProperty("b", "42");
+     *     TStream<JsonObject> rc = HttpStreams.postJson(
+     *         topology.collection(Arrays.asList(request1)),
+     *         HttpClients::noAuthentication,
+     *         t -> url,
+     *         t -> new ByteArrayEntity(request1.toString().getBytes())
+     *     );
+     * }
+     * </pre>
+     * <br>
+     * See HttpTest for example.
+     * <br>
+     * @param stream - JsonObject TStream.
+     * @param clientCreator - CloseableHttpClient supplier preferably created using {@link HttpClients}
+     * @param uri - URI function which returns URI string
+     * @param body - Function that returns HttpEntity which will be set as a body for the request.
+     * @return TStream of JsonObject which contains responses of GET requests
+     * 
+     * @see HttpStreams#requestsWithBody(TStream, Supplier, Function, Function, Function, BiFunction)
+     */
+    public static TStream<JsonObject> postJson(TStream<JsonObject> stream,
+            Supplier<CloseableHttpClient> clientCreator,
+            Function<JsonObject, String> uri,
+            Function<JsonObject, HttpEntity> body) {
```

Comment:
It strikes me that the ``body`` function should instead be ``UnaryOperator<JsonObject>`` (a fn that yields a JsonObject (body) from the input JsonObject tuple).  I think that's more in line with "keeping the simple simple" for applications exchanging Json-based data.  I agree that ``HttpEntity`` is appropriate for ``requestsWithBody()``.  Does that make sense?